### ISSUE TITLE
Variable MC samples specified with placeholders causes issues for Reshape layer

### ISSUE THREAD

**Issue by ltiao** [data61/aboleth#160]
```
Traceback (most recent call last):
  File "mnist_softmax_regression.py", line 112, in <module>
    main()
  File "mnist_softmax_regression.py", line 79, in main
    logits, reg = net(X=X)
  File "/home/tia00c/projects/aboleth/aboleth/baselayers.py", line 69, in __call__
    Net, KL = self._build(**kwargs)
  File "/home/tia00c/projects/aboleth/aboleth/baselayers.py", line 98, in _build
    Net, KL = self.stack(**kwargs)
  File "/home/tia00c/projects/aboleth/aboleth/baselayers.py", line 159, in stackfunc
    result1, loss1 = layer1(*args, **kwargs)
  File "/home/tia00c/projects/aboleth/aboleth/baselayers.py", line 159, in stackfunc
    result1, loss1 = layer1(*args, **kwargs)
  File "/home/tia00c/projects/aboleth/aboleth/baselayers.py", line 159, in stackfunc
    result1, loss1 = layer1(*args, **kwargs)
  [Previous line repeated 7 more times]
  File "/home/tia00c/projects/aboleth/aboleth/baselayers.py", line 160, in stackfunc
    result, loss2 = layer2(result1)
  File "/home/tia00c/projects/aboleth/aboleth/baselayers.py", line 32, in __call__
    Net, KL = self._build(X)
  File "/home/tia00c/projects/aboleth/aboleth/layers.py", line 236, in _build
    new_shape = (int(X.shape[0]), tf.shape(X)[1]) + self.target_shape
TypeError: __int__ returned non-int (type NoneType)
```

**Comment by dsteinberg** [comment]
This would definitely be caught by #147

**Comment by dsteinberg** [comment]
fixed in #192
